plugins/quetz_sql_authenticator/tests/test_quetz_sql_authenticator.py

```diff
@@ -0,0 +1,129 @@
+from contextlib import contextmanager
+from unittest import mock
+
+from click.testing import CliRunner
+from quetz_sql_authenticator.cli import _calculate_hash, _create, _delete, _reset
+from quetz_sql_authenticator.db_models import Credentials
+
+
+# Test invalid login
+def test_invalid_login(client):
+    response = client.post(
+        "/auth/sql/authorize",
+        data={"username": "testuser", "password": "testpassword"},
+    )
+    # Unauthorized
+    assert response.status_code == 401
+    assert "login failed" in response.text
+
+
+# Test valid login
+def test_valid_login(client, db):
+    # Insert user
+    credentials = Credentials(
+        username="testuser", password=_calculate_hash("testpassword")
+    )
+    db.add(credentials)
+
+    response = client.post(
+        "/auth/sql/authorize",
+        data={"username": "testuser", "password": "testpassword"},
+    )
+    # Assert that we get a redirect to the main page
+    assert response.status_code == 303
+
+
+# Test create user
+def test_create_user(db):
+    @contextmanager
+    def get_db():
+        yield db
+
+    runner = CliRunner()
+    with mock.patch("quetz_sql_authenticator.cli.get_db_manager", get_db):
+        result = runner.invoke(_create, ["testuser", "testpassword"])
+        assert result.exit_code == 0
+        assert "INFO: User 'testuser' created successfully." in result.output
+        # Check that user is in table
+        assert (
+            db.query(Credentials)
+            .filter(Credentials.username == "testuser")
+            .one_or_none()
+            is not None
+        )
+
+
+# Test create user
+# def test_create_user_api(client, db):
+#     response = client.get(
+#         f"/api/channels/a/b/conda-suggest"
+#     )
+#     print(response.url)
+#     # Unauthorized
+#     assert "login failed" in response.text
+
+#     # Check that user is in table
+#     assert (
+#         db.query(Credentials)
+#         .filter(Credentials.username == "testuser")
+#         .one_or_none()
+#         is not None
+#     )
+
+
+def test_conda_suggest_endpoint_without_upload(client):
+    response = client.get(f"/api/hello")  # noqa
+    assert response.status_code == 200
```

Comment:
@simonbohnen, [wrt to your comment about failing tests](https://github.com/mamba-org/quetz/pull/508#issuecomment-1286682190): I'm not quite sure what you mean. The only test that is failing for me is this one and this seems misplaced. can you explain in more detail what's not working for you?

Reply:
Sorry, I didn't word my previous comment well.
- I'm trying to register API routes for this plugin
- They work correctly when launching `quetz start`, e.g. when sending a simple get to `/api/hello`
- They **don't** work in the tests, they [always return `404`](https://github.com/mamba-org/quetz/actions/runs/3296153866/jobs/5438042723#step:7:903)
- That's why I added this sample test, which I forgot to rename
- The tests for the conda-suggest plugin are obviously working
- I tried to figure out how conda-suggest registers their routes, and found [this](https://github.com/janjagusch/quetz/blob/9250ef950ca988a15761c112e3f848bb58de6fa4/plugins/quetz_sql_authenticator/quetz_sql_authenticator/main.py#L11)
- Apparently that's not sufficient somehow, and that's where I'm stuck currently

Reply:
strange, indeed. I had a quick look comparing this vs. the conda-suggest plugin, and couldn't find anything obvious.

Reply:
See https://github.com/mamba-org/quetz/pull/508/commits/db8c247001bb1fa2beab6ec703d4eff9361296ed. We had to set the `plugins` fixture.